Enterprise-CMCS/eAPD
# Nz/4354 priorities scope

Resolves #4354

### Description
Creating fields for the `State Priorities and Scope of APD` page, with validation, and connection to backend.

### Significant changes or possible side effects
- add fields to State Priorities and Scope of (MMIS) APD page
- add validation to fields
- connect fields to backend

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| State Priorities and Scope of APD Page | On the page | Should have all the correct fields in the right order | cypress, jest |
| State Priorities and Scope of APD Page | Admin check is on | If field is blank, there should be validation error message under text box | jest |
| State Priorities and Scope of APD Page: Medicaid Program and Priorities field | Admin check is on | If field is blank, the validation error message should be: `Provide Medicaid Program and Priorities` | cypress |
| State Priorities and Scope of APD Page: Medicaid Enterprise System Introduction field | Admin check is on | If field is blank, the validation error message should be: `Provide a Medicaid Enterprise System Introduction` | cypress |
| State Priorities and Scope of APD Page: Scope of APD field | Admin check is on | If field is blank, the validation error message should be: Provide an Overview of the APD's Scope | cypress |

### Steps to manually verify this change

1. Create or go to an MMIS APD
2. Go to `State Priorities and Scope` in the side nav
3. Confirm all fields are there in right order
4. Can input text in all fields and it will save
5. On reload the text stays
6. When `Admin Check` is on and fields are blank, correct errors show

### This pull request is ready to code review when

      tests, refactor
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when


### This pull request is ready to review when the QA has


### This pull request can be merged when
